### PR TITLE
Add `cupy.union1d` API.

### PR DESCRIPTION
**Pull request by khushi-411**
Hi, Team!!
The PR **aims** to add `cupy.union1d` API. It follows #6078.
I would love to get your feedback.

Thanks! :)

### CODE REVIEW

File: tests/cupy_tests/logic_tests/test_truth.py [hunk 1]
```diff
@@ -123,3 +125,31 @@ def test(self, xp, dtype):
...
+    @testing.numpy_cupy_array_equal()
+    def test_union1d_2(self, xp, dtype):
+        x = testing.shaped_arange((5, 2), xp, dtype=dtype)
+        y = testing.shaped_arange((2, 3, 4), xp, dtype=dtype)
+        return xp.union1d(x, y)
+
+    @testing.for_all_dtypes()
+    def test_union1d_3(self, dtype):
+        for xp in (numpy, cupy):
+            x = xp.array([0+1j, 3+2j], dtype=dtype)
```
Comment (takagi) [inline tests/cupy_tests/logic_tests/test_truth.py:150]: This line causes an error: `TypeError: float() argument must be a string or a real number, not 'complex'`. Would you look again this test method?

File: tests/cupy_tests/logic_tests/test_truth.py [hunk 2]
```diff
@@ -123,3 +125,31 @@ def test(self, xp, dtype):
             return xp.in1d(x, y, self.assume_unique, self.invert)\
                 .reshape(x.shape)
         return getattr(xp, self.f)(x, y, self.assume_unique, self.invert)
+
+
+@testing.gpu
```
Comment (takagi) [inline tests/cupy_tests/logic_tests/test_truth.py:130]:
```suggestion
```

**Comment by takagi** [comment]
Thanks for your work! It's almost fine except for a test method I mentioned above.

File: tests/cupy_tests/logic_tests/test_truth.py (hunk 1 above)
Comment (khushi-411) [reply tests/cupy_tests/logic_tests/test_truth.py:149]: Yes, sure! I will correct it and ping you soon :)
Thanks!

**Comment by khushi-411** [comment]
Hi, @takagi!
Thanks a lot for reviewing! :)

That test case was wrong (I had a different thing in my mind), anyways I made the required corrections :) Thank you!

Will you please take a look? Thanks!

**Review by takagi — APPROVED** [review]

**Comment by takagi** [comment]
LGTM! Waiting for CI pass to merge.

**Pull request merged**